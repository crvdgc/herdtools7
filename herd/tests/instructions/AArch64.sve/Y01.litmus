AArch64 Y01
Variant=sve
{
uint32_t t[8] = {1,2,3,4,5,6,7,8};
uint32_t s[8] = {8,7,6,5,4,3,2,1};
0:X1=t;
0:X2=s;
}

(* Adapted from V25 *)
(* This test does not depends on scalable vector length *)

  P0                  ;
CNTW X0               ;
CMP X0,#8             ;
B.GT L3               ;
MOV X3,#0             ;
PTRUE P0.S            ;
L1:                   ;
LD1W {Z0.S},P0/Z,[X1] ;
LD1W {Z1.S},P0/Z,[X2] ;
EOR Z0.S,Z0.S,Z1.S    ;
ST1W {Z0.S},P0,[X1]   ;
ADD X1,X1,X0,LSL #2   ;
ADD X2,X2,X0,LSL #2   ;
INCW X3               ;
L2:                   ;
CMP X3,#8             ;
B.LT L1               ;
B L0                  ;
L3:                   ;
PTRUE P0.S,VL8        ;
LD1W {Z0.S},P0/Z,[X1] ;
LD1W {Z1.S},P0/Z,[X2] ;
EOR Z0.S,Z0.S,Z0.S    ;
ST1W {Z0.S},P0,[X1]   ;
L0:                   ;
forall (
  t={9,5,5,1,1,5,5,9} /\
  s={8,7,6,5,4,3,2,1}
)

