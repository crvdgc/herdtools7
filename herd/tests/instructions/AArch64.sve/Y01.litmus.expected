Test Y01 Required
States 1
s={8,7,6,5,4,3,2,1}; t={9,5,5,1,1,5,5,9};
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Vector-Extensions-is-work-in-progress
Condition forall (t={9,5,5,1,1,5,5,9} /\ s={8,7,6,5,4,3,2,1})
Observation Y01 Always 1 0
Hash=8f987c02b299c6f220d7903ec693f66a

